src/auth/baseexternalclient.ts

```diff
@@ -715,4 +719,69 @@ export abstract class BaseExternalAccountClient extends AuthClient {
       : 'unknown';
     return `gl-node/${nodeVersion} auth/${pkg.version} google-byoid-sdk source/${credentialSourceType} sa-impersonation/${saImpersonation} config-lifetime/${this.configLifetimeRequested}`;
   }
+
+  /**
+   * Returns the workforce identity pool-id if it is determinable
+   * from the audience resource name.
+   * @param audience The STS audience used to determine the pool-id.
+   * @return The pool-id associated with the workforce identity pool, if
+   *   this can be determined from the STS audience field. Otherwise, null is
+   *   returned.
+   */
+  #getWorkForcePoolId(audience: string): string | null {
+    // STS audience pattern:
+    // .../workforcePools/$WORKFORCE_POOL_ID/providers/...
+    return (
+      audience.match(/\/workforcePools\/(?<workforcePool>[^/]+)\/providers\//)
+        ?.groups?.workforcePool ?? null
+    );
+  }
+
+  /**
+   * Returns the workload identity pool-id if it is determinable
+   * from the audience resource name.
+   * @param audience The STS audience used to determine the pool-id.
+   * @return The pool-id associated with the workload identity pool, if
+   *   this can be determined from the STS audience field. Otherwise, null is
+   *   returned.
+   */
+  #getWorkloadPoolId(audience: string): string | null {
+    // STS audience pattern:
+    // .../workloadIdentityPools/POOL_ID/providers/...
+    return (
+      audience.match(
+        /\/workloadIdentityPools\/(?<workloadPool>[^/]+)\/providers\//,
+      )?.groups?.workloadPool ?? null
+    );
+  }
+
+  getTrustBoundaryUrl(): string {
+    let lookupTrustBoundaryUrl = null;
+    if (this.audience.match(WORKFORCE_AUDIENCE_PATTERN)) {
+      //client configured for workforce authorization
+      const wfPoolId = this.#getWorkForcePoolId(this.audience);
+      if (!wfPoolId) {
+        throw new RangeError(
+          'TrustBoundaryLookup: Failed to fetch trust boundary data due to missing workforce pool id',
```

Comment:
These errors should just tell the user that the audience they provided is invalid. Same below.